### PR TITLE
Classify AR Foundation provider as experimental, fix for 6461

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change addresses #6461 by checking to see if the xr settings device name is null or empty when running on android. it also removes the async check for device support.

At the same time, the ar foundation camera settings provider is being moved into the extensions package under a new providers folder. Additionally, it is being placed in an experimental folder while the provider gains community exposure and feedback.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset [hunk 1]
```diff
@@ -26,15 +26,15 @@ MonoBehaviour:
   lightingScenes:
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset]: This change seems unrelated?

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs [hunk 2]
```diff
@@ -106,21 +108,16 @@ private void FindARFoundationComponents()
...
             base.Initialize();
 
-#if !(WINDOWS_UWP && !ENABLE_IL2CPP)
-            ARSessionState sessionState = (ARSessionState)(await ARSession.CheckAvailability());
-            if (ARSessionState.Ready > sessionState)
+            if (Application.platform == RuntimePlatform.Android)
             {
-                Debug.LogError("Unable to initialize the Unity AR Camera Settings provider. Device support for AR Foundation was not detected.");
-                isInitialized = true;
+                isSupportedConfiguration = string.IsNullOrWhiteSpace(XRSettings.loadedDeviceName);
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: Would be good to add a comment here to explain what it's doing

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs [hunk 3]
```diff
@@ -79,9 +80,10 @@ public UnityARCameraSettingsProfile SettingsProfile
             }
         }
 
+#if !(WINDOWS_UWP && !ENABLE_IL2CPP)
+        bool isSupportedConfiguration = true;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: missing access modifiers?

File: Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealitySceneSystemProfile.asset [hunk 4]
```diff
@@ -26,8 +26,8 @@ MonoBehaviour:
   lightingScenes:
   - Name: DefaultLightingScene
     Path: Assets/MixedRealityToolkit.SDK/StandardAssets/Scenes/DefaultLightingScene.unity
-    Included: 1
-    BuildIndex: 1
+    Included: 0
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealitySceneSystemProfile.asset]: This change seems unrelated

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealitySceneSystemProfile.asset]: correct, i'll put the current ones back. nice catch!

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs [hunk 5]
```diff
@@ -79,9 +81,10 @@ public UnityARCameraSettingsProfile SettingsProfile
             }
         }
 
-        bool isInitialized = false;
-
 #if !(WINDOWS_UWP && !ENABLE_IL2CPP)
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: Do we still need these with C# Projects turned off?

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: In theory, yes. Since Unity would compile the asmdefs using the .net backend, correct?

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: It shouldn't, since the asmdef is marked as iOS and Android only

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: oh yeah. i see where you are going now :)

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: Might need to instead wrap it in iOS and Android checks though, since it's marked as valid in the Editor as well.
That way, devs who only consume MRTK via the Windows targets aren't required to import the ARFoundation package if they install our extensions.

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: I also wonder if we should redefine the UnityARCameraSettingsProfile a bit, so we don't directly serialize ARFoundation types? And wrapping serialized fields in `#if` is what leads to https://github.com/microsoft/MixedRealityToolkit-Unity/issues/6054 and https://github.com/microsoft/MixedRealityToolkit-Unity/issues/3971

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs (hunk 2 above)
Comment (wiwei) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: I think that it would still be helpful to have a comment about the significant of the empty loadedDeviceName.

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs [hunk 6]
```diff
@@ -79,9 +81,10 @@ public UnityARCameraSettingsProfile SettingsProfile
...
+        private bool isSupportedConfiguration = true;
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: If this class is both an AR foundation and non-AR foundation handling thing, and this supported configuration is about ARFoundation only, it would be worth naming it as such (Or having a comment that this is about AR foundation only)

Comment (wiwei) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: Otherwise it feels like this should gate the functionality of the entire class.

File: Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityCameraProfile.asset [hunk 7]
```diff
@@ -13,14 +13,7 @@ MonoBehaviour:
   m_Name: DefaultMixedRealityCameraProfile
   m_EditorClassIdentifier: 
   isCustomProfile: 0
-  settingsConfigurations:
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityCameraProfile.asset]: Since we're no longer having this by default, are there any docs that explain how to set this up?

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/Definitions/ArTrackedPose.cs [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        /// </summary>
+        LeftPose = 4,
+
+        /// <summary>
+        /// The right hand controller pose.
+        /// </summary>
+        RightPose = 5,
+
+        /// <summary>
+        /// The color camera of a movible (ex: phone) device.
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/Definitions/ArTrackedPose.cs]:
```suggestion
        /// The color camera of a movable (ex: phone) device.
```

File: Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityCameraProfile.asset (hunk 7 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityCameraProfile.asset]: there will be. i am writing docs after we branch for stabilization (they do not risk breaking the build) :)

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs [hunk 9]
```diff
@@ -188,9 +189,12 @@ private void InitializeARFoundation()
             if (SettingsProfile != null)
             {
                 // Read settings to be applied to the camera.
-                poseSource = SettingsProfile.PoseSource;
-                trackingType = SettingsProfile.TrackingType;
-                updateType = SettingsProfile.UpdateType;
+                // The enums used by the profile use the same values as those provided by Unity.
+                // We use custom enums to avoid customers needing to include packages and/or namespaces simply to resolve
+                // enum values in the profile.
+                poseSource = (TrackedPoseDriver.TrackedPose)SettingsProfile.PoseSource;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: Do we want to run switches here, in case the order / numbering of the Unity enums change?

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs (hunk 6 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: i can rename the variable -> isSupportedArConfiguration

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs (hunk 2 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: i thought i added a comment.... will confirm

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs (hunk 5 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: done

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/Definitions/ArTrackedPose.cs (hunk 8 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/Definitions/ArTrackedPose.cs]: was supposed to be mobile :D

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs (hunk 9 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: we could

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs (hunk 2 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: clarified comment a bit

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/Definitions/ArTrackedPose.cs (hunk 8 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/Definitions/ArTrackedPose.cs]: fixed typing error :)

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/Definitions/ArTrackedPose.cs]: Ah!

File: Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityCameraProfile.asset (hunk 7 above)
Comment (wiwei) [reply Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityCameraProfile.asset]: Going forward it would be worth including the docs with the code change - per our PR guidelines, we should uphold that in all of our PRs:

https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/PullRequests.html#documentation-should-be-added-in-the-same-pull-request-as-a-fix--feature
